manifest: Update Matter SDK revision to pull the newest changes (#14128)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

  To :ref:`inherit Thread certification <ug_matter_device_certification_reqs_dependent>` from Nordic Semiconductor, you must use the PSA Crypto API backend.

* Updated:

  * The :ref:`ug_matter_device_low_power_configuration` page with the information about Intermittently Connected Devices (ICD) configuration.
  * The Factory Data Guide regarding the changes for the SPAKE2+ verifier, and generation of the Unique ID for Rotating Device ID.

* Changed the factory data default usage:

  * The SPAKE2+ verifier is now generated by default with each build, and it will change if any of the related Kconfig options are modified.
    This resolves the :ref:`known issue <known_issues>` related to the SPAKE2+ verifier not regenerating (KRKNWK-18315).
  * The Test Certification Declaration can now be generated independently from the generation of the DAC and PAI certificates.
  * The Unique ID for Rotating Device ID can now only be generated if the :kconfig:option:`CONFIG_CHIP_ROTATING_DEVICE_ID` Kconfig option is set to ``y``.

* Added:

  * A Kconfig option for disabling or enabling :ref:`ug_matter_configuring_read_client`.

>   * The section `Default Kconfig values and developing aspects` to the Factory Data Guide that describes useful developer tricks and device configuration.

Reviewer: ```suggestion
  * The Default Kconfig values and developing aspects section to the :doc:`matter:nrfconnect_factory_data_configuration` page.
    The section contains useful developer tricks and device configurations.
```

* For devices that use Matter over Thread, the default cryptography backend is now Arm PSA Crypto API instead of Mbed TLS, which was used in earlier versions.
  You can still build all examples with deprecated Mbed TLS support by setting the :kconfig:option:`CONFIG_CHIP_CRYPTO_PSA` Kconfig option to ``n``, but you must build the Thread libraries from sources.
  To :ref:`inherit Thread certification <ug_matter_device_certification_reqs_dependent>` from Nordic Semiconductor, you must use the PSA Crypto API backend.

* Updated:

  * The :ref:`ug_matter_device_low_power_configuration` page with the information about Intermittently Connected Devices (ICD) configuration.

>   * The Factory Data Guide and added a new section explaining several important Kconfig options.

Author: I have no idea how to link a Github .md file section in our docs :) Maybe @wiba-nordic do you know that? I can describe it more.

>   * The Factory Data Guide and added a new section explaining several important Kconfig options.

Reviewer 2: I think it could be done similarly to how we link :doc:`matter:nrfconnect_examples_software_update` in our sample's README.rst

* Changed the factory data default usage:

>   * The SPAKE2+ verifier is now generated by default with each build, and it will change if any of the related Kconfig options are modified.

Reviewer 3: I think we need to remove now the entry in known-issue about this issue.

Reviewer: Wouldn't this just mean that we don't update the issue as affecting any upcoming versions? It still affects the v2.5-based releases, so should remain in the known issues.

Reviewer 3: Ok, that makes sense!

    This resolves the :ref:`known issue <known_issues>` related to the SPAKE2+ verifier not regenerating (KRKNWK-18315).
  * The Test Certification Declaration can now be generated independently and no longer necessitates the generation of the DAC and PAI certificates.

>   * The Rotating Device ID Unique ID can be utilized and generated only if the :kconfig:option:`CONFIG_CHIP_ROTATING_DEVICE_ID` Kconfig option is set to ``y``.

Reviewer 3: Can you ellaborate more on this point. From description i don't understand what has changed, and RDID UID could be utilized and generated before. Maybe we can mention, that some KConfig is no longer needed now or sth.

Reviewer: We could change this line to:
```suggestion
  * The Rotating Device ID Unique ID can now only be utilized and generated if the :kconfig:option:`CONFIG_CHIP_ROTATING_DEVICE_ID` Kconfig option is set to ``y``.
```
Adding "now" and moving "only" up a bit should make it clearer.

>   * The Rotating Device ID Unique ID can now only be utilized and generated if the :kconfig:option:`CONFIG_CHIP_ROTATING_DEVICE_ID` Kconfig option is set to ``y``.

Reviewer 2: ```suggestion
  * The Unique ID for Rotating Device ID can now only be generated if the :kconfig:option:`CONFIG_CHIP_ROTATING_DEVICE_ID` Kconfig option is set to ``y``.
```

  You can still build all examples with deprecated Mbed TLS support by setting the :kconfig:option:`CONFIG_CHIP_CRYPTO_PSA` Kconfig option to ``n``, but you must build the Thread libraries from sources.
  To :ref:`inherit Thread certification <ug_matter_device_certification_reqs_dependent>` from Nordic Semiconductor, you must use the PSA Crypto API backend.

> * Updated the page about :ref:`ug_matter_device_low_power_configuration` with the information about Intermittently Connected Devices (ICD) configuration.
> * Updated the Factory Data Guide and added a new section explaining several important Kconfig options.
> 
> * Refactored the factory data default usage:
> 
>   * The SPAKE2+ verifier is now generated by default with each build, and it will change if any of the related Kconfig options are modified.
>     This resolves the known issue: KRKNWK-18315.
>   * The Test Certification Declaration can now be generated independently and no longer necessitates the generation of the DAC and PAI certificates.
>   * The Rotating Device ID Unique ID can be utilized and generated only if the :kconfig:option:`CONFIG_CHIP_ROTATING_DEVICE_ID` Kconfig option is set to ``y``.

Reviewer: ```suggestion
* Updated:

   * The :ref:`ug_matter_device_low_power_configuration` page with the information about Intermittently Connected Devices (ICD) configuration.
  * The Factory Data Guide and added a new section explaining several important Kconfig options.

* Changed the factory data default usage:

  * The SPAKE2+ verifier is now generated by default with each build, and it will change if any of the related Kconfig options are modified.
    This resolves the :ref:`known issue <known_issues>` related to the SPAKE2+ verifier not regenerating (KRKNWK-18315).
  * The Test Certification Declaration can now be generated independently and no longer necessitates the generation of the DAC and PAI certificates.
  * The Rotating Device ID Unique ID can be utilized and generated only if the :kconfig:option:`CONFIG_CHIP_ROTATING_DEVICE_ID` Kconfig option is set to ``y``.
```
